### ISSUE TITLE
Chunking of subrequests

### ISSUE THREAD

**Issue by DuckDagobert** [openresty/echo-nginx-module#49]
Hey

For echo_flush it states:

> This directive will fail to flush the output buffer in case of subrequests get involved

Does chunking subrequests work however with this module (on nginx 1.5+; chunking is part of nginx core since 1.3.9)? (no need for echo_flush then)

**Comment by agentzh** [comment]
In nginx, "flushing" has nothing to do with "chunking". So I'm confused here. Also, you seem to be confused chunked response bodies with chunked request bodies. Anyway, I don't quite follow you.

**Comment by DuckDagobert** [comment]
You are right, sorry this was a bit disambigous (and I had a little confusion).
What I really mean:
If the subrequest (called via echo_location_async) includes the echo_flush, will it flush it, or still wait for the other subrequests?
If I call php in a subrequest block (called via echo_location_async) and use the flush() (and ob_flush) in the php file, will it flush it or not? (so does it keep the "normal" behavior of php or not)

**Comment by agentzh** [comment]
@DuckDagobert If there is no pending subrequests to complete before the current subrequest, then my hunch is that `echo_flush` should do what it is supposed to do. But you'll have to do your own testing to be sure. The PHP thing can be more complicated, depending on how you configure your ngx_fastcgi module's buffering and your PHP implementation's buffering. Oh well.

**Comment by DuckDagobert** [comment]
After some testing and some more research, I decided to do it all with lua nginx module (as I have more control, need fewer subrequests, can have non blocking subrequests and can echo all subrequests separately, optimised code,... using a couple openresty module; basically emulating facebook's bigpipe/how I think taobao does their pagelets)

You really are a genius! openresty is amazing :-)
